File: dao-lesson/src/main/java/com/acme/recipes/data/util/DatabaseInitTransaction.java
package com.acme.recipes.data.util;

import com.acme.recipes.data.dao.RecipeDao;
import com.acme.recipes.data.entity.IngredientEntity;
import com.acme.recipes.data.entity.RecipeEntity;

import java.util.Arrays;
import java.util.UUID;

import javax.annotation.Nonnull;

import io.realm.Realm;

public class DatabaseInitTransaction implements Realm.Transaction {

    @Override
    public void execute(@Nonnull Realm db) {

        RecipeDao recipeDao = new RecipeDao(db);

        db.deleteAll();

        RecipeEntity eclair = recipeDao.copyOrUpdate(
            new RecipeEntity(UUID.randomUUID().toString(),
                    "Eclair",
                    "Pastry filled with cream",
                    800)
        );
        eclair.getIngredients().addAll(
            Arrays.asList(
                new IngredientEntity("Sugar",1,  "cup"),
                new IngredientEntity("Flour",1.5,"cup"),
                new IngredientEntity("Cocoa",1,  "tbsp"),
                new IngredientEntity("Milk", 1,  "quart")
            )
        );

        RecipeEntity oreo = recipeDao.copyOrUpdate(
            new RecipeEntity(UUID.randomUUID().toString(),
                    "Oreo",
                    "Chocolate cookie",
                    650)
        );
        oreo.getIngredients().addAll(
            Arrays.asList(
                new IngredientEntity("Sugar",1,"cup"),
                new IngredientEntity("Flour",2,"cup")
            )
        );

    }

}

File: app/src/main/java/com/acme/recipes/database/util/DatabaseInitTransaction.kt
package com.acme.recipes.database.util

import com.acme.recipes.database.dao.RecipeDao
import com.acme.recipes.database.entity.IngredientEntity
import com.acme.recipes.database.entity.RecipeEntity
import io.realm.Realm
import java.util.*

class DatabaseInitTransaction : Realm.Transaction {

    override fun execute(db: Realm) {

        val recipeDao = RecipeDao(db)

        db.deleteAll()

        val eclair = recipeDao.copyOrUpdate(
                RecipeEntity(UUID.randomUUID().toString(),
                        "Eclair",
                        "Pastry filled with cream",
                        800)
        )
        eclair.ingredientEntities.addAll(
                listOf(
                        IngredientEntity("Sugar", 1.0, "cup"),
                        IngredientEntity("Flour", 1.5, "cup"),
                        IngredientEntity("Cocoa", 1.0, "tbsp"),
                        IngredientEntity("Milk", 1.0, "quart")
                )
        )

        val oreo = recipeDao.copyOrUpdate(
                RecipeEntity(UUID.randomUUID().toString(),
                        "Oreo",
                        "Chocolate cookie",
                        650)
        )
        oreo.ingredientEntities.addAll(
                listOf(
                        IngredientEntity("Sugar", 1.0, "cup"),
                        IngredientEntity("Flour", 2.0, "cup")
                )
        )

    }

}
